dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -235,8 +299,251 @@ public void Test_Constant_fields_are_in_map() throws Exception {
+                new TestCase(BaseContentType.getBaseContentType(5),
+                        contentletAPI.search("+basetype:5", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(6),
+                        contentletAPI.search("+basetype:6", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(7),
+                        contentletAPI.search("+basetype:7", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(8),
+                        contentletAPI.search("+basetype:8", limit, 0, null, user, respectFrontEndUsers))
+        };
+    }
+
+    private static class TestCase {
+        final BaseContentType baseContentType;
+        final List<Contentlet> contentlets;
+
+        TestCase(final BaseContentType baseContentType,
+                final List<Contentlet> contentlets) {
+            this.baseContentType = baseContentType;
+            this.contentlets = contentlets;
+        }
+    }
+
+    @Test
+    @UseDataProvider("listTestCases")
+    public void Transformer_Backwards_Compatibility_Test(final TestCase testCase) {
+
+        final List <Contentlet> list = testCase.contentlets;
+
+        if(list.isEmpty()){
+           Logger.warn(ContentletTransformerTest.class, String.format("Unable to get test samples of type `%s`",testCase.baseContentType.name()));
+           return;
+        }
+
+        final String baseTypeName = testCase.baseContentType.name();
+
+        final ContentletToMapTransformer contentletToMapTransformer = new ContentletToMapTransformer(list);
+
+        final List<Map<String, Object>> transformedList1 = contentletToMapTransformer.toMaps();
+
+        final DotTransformer dotTransformer = new DotTransformerBuilder().defaultOptions().content(list).build();
+
+        final List<Map<String, Object>> transformedList2 = dotTransformer.toMaps();
+
+        assertEquals(transformedList1.size(), transformedList2.size());
+        for (int i = 0; i < list.size(); i++) {
+
+            final Contentlet original = list.get(i);
+            //We compare the transformation results are the same using the default options on the new Transformer
+            final Map<String, Object> objectMap1 = transformedList1.get(i);
+
+            final Map<String, Object> objectMap2 = transformedList2.get(i);
+
+            assertTrue(String.format(" baseType `%s` should have same (or more) number of properties " ,baseTypeName),objectMap2.size() >= objectMap1.size());
+
+            for (final String key1 : objectMap1.keySet()) {
+
+                final Object object1 = objectMap1.get(key1);
+                final Object object2 = objectMap2.get(key1);
+
+                assertNotNull(object2);
+
+                if( testCase.baseContentType == BaseContentType.FILEASSET  &&  "title".equals(key1)){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)